### PR TITLE
Docker compose frontend

### PR DESCRIPTION
**Pull request by bmath10**
**Note:** This PR updates how docker builds and runs the front-end so it will be easier to add to the back-end's docker-compose.
- The Dockerfile now builds the project instead of relying on `ng build` to run before building the docker image. This was required in order to add this project to the back-end's docker-compose by pulling this repo.
- Certs are required in order to get yarn to run with the proxy. Like the instructions for the back-end, add a `certs` directory at the root of the project and add your certs using Matt's pinned instructions on the slack channel.
- To test: run using the other docker instructions

 **UPDATE 12/18** 

This branch was rebased off of master to add the config stuff. Make sure that:
- the node modules are updated (run `yarn install`)
- either `env.sample` is copied to `.env` or the LANTERN_GRAFANA_URL variable is set directly

--------
Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
- [x] Administrative tasks are complete.
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-93
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
- [x] Code review has been performed.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code accomplishes the tasks purpose.
- [x] Tests are complete.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge is complete.

### CODE REVIEW

**Review by mrnosal — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -34,14 +46,65 @@ To run with the nginx server (on a Mac):
...
+```
+
+To stop the service, remove the container and image, run:
+
+```bash
+docker-compose down --rmi all -v
+```
+
+### Starting Services Behind SSL-Inspecting Proxy
+If you are operating behind a proxy that does SSL-Inspection, yarn needs to be configured with the certificates that are used by the proxy in the image. Currently, `docker-compose.yml` includes the location of a `certs` directory as an argument to the `Dockerfile`. If you are operating behind an SSL-Inspecting proxy **you will have to copy your certificates into this directory.**
```
Comment (mayerm94) [inline README.md]: You should mention that it is expected that the root certificate will exist in a file named `ca-certificates.crt` since there is the line in the Dockerfile that expects that file to exist

File: README.md [hunk 2]
```diff
@@ -35,13 +44,26 @@ To stop the server:
...
-1. Build the production version of this project: `ng build --prod`
-2. Build image: `docker build -t angular-nginx .`
-3. Run container: `docker run --name angular-nginx-container -d -p 8090:80 angular-nginx`
-4. Go to `localhost:8090`
+To run without using the docker-compose command:
+
+First build the image:
+
+```bash
+docker build --build-arg cert_dir=./certs -t lantern_front_end .
```
Comment (mayerm94) [inline README.md:64]: Can you add the command used to publish the image to dockerhub?

Comment (bmath10) [reply README.md:64]: 👍

File: README.md (hunk 1 above)
Comment (bmath10) [reply README.md]: 👍

**Review by mayerm94 — DISMISSED** [review]
👍 Just needs a rebase

**Review by mayerm94 — APPROVED** [review]
👍

**Review by mrnosal — APPROVED** [review]
